packages/client-sdk-web/src/internal/webhook-client.ts

```diff
@@ -207,4 +209,44 @@ export class WebhookClient implements IWebhookClient {
       );
     });
   }
+
+  async rotateWebhookSecret(
+    id: WebhookId
+  ): Promise<RotateWebhookSecret.Response> {
+    try {
+      validateCacheName(id.cacheName);
+      validateWebhookName(id.webhookName);
+    } catch (err) {
+      return new RotateWebhookSecret.Error(normalizeSdkError(err as Error));
+    }
+
+    const request = new _RotateWebhookSecretRequest();
+    const webhookId = new _WebhookId();
+    webhookId.setWebhookName(id.webhookName);
+    webhookId.setCacheName(id.cacheName);
+    request.setWebhookId(webhookId);
+    this.logger.debug('issuing "RotateWebhookSecret" request');
+
+    return await new Promise<RotateWebhookSecret.Response>(resolve => {
+      this.webhookClient.rotateWebhookSecret(
+        request,
+        this.clientMetadataProvider.createClientMetadata(),
+        (err, resp) => {
+          if (err || !resp) {
+            resolve(
+              new RotateWebhookSecret.Error(cacheServiceErrorMapper(err))
+            );
+          } else {
+            resolve(
+              new RotateWebhookSecret.Success({
+                secret: resp.getSecretString(),
+                webhookName: id.webhookName,
```

Comment:
the protos don't include this, but I think it might be useful to include in the response. If we want to keep these 1:1 with the protos, I can remove it here.

Reply:
nah i'm on board